src/utils.ts

```diff
@@ -436,3 +445,67 @@ export function findProfiledTransactionsFromEnvelope(envelope: Envelope): Event[
 
   return events;
 }
+
+const debugIdStackParserCache = new WeakMap<StackParser, Map<string, StackFrame[]>>();
```

Comment:
Generally, this looks good to me for now. I am wondering if we could generalize this solution in the JS sdk directly by applying the debug metadata also to profiling events.